### ISSUE TITLE
Change vtkCutter in slice wigets

### ISSUE THREAD

**Issue by gbene** [pyvista/pyvista#5559]
### Describe the feature you would like to be added.

Hi all!

I was playing around with the `add_mesh_slice` widget and found out that it could be improved. In VTK 9.3 a new `vtkStructuredDataPlaneCutter` algorithm has been introduced. This is quite faster than the classic `vtkCutter` algorithm because of this could improve drastically the time needed to slice a given mesh. For example with this cutter we could use the widget in a continuous way (i.e. interaction_event='always'). 

This is an example with the standard `vtkCutter` :

https://github.com/pyvista/pyvista/assets/40857804/59f69a12-ed08-4649-b10e-4a4dfeec7e62

This is the same with the `vtkStructuredDataPlaneCutter`:

https://github.com/pyvista/pyvista/assets/40857804/d508d876-dc3f-46f8-b2d2-0873422ca9f5

As you can see it is much quicker!
The only problem is that flickering red surface of the widget that is quite annoying and I could not figure out how to remove it.

As a bonus here is the same example with  `vtkStructuredDataPlaneCutter` but using a slider widget (a way to remove the red flickering):

https://github.com/pyvista/pyvista/assets/40857804/1872521c-642c-4c61-8326-dfecb8ae1e23

Changes to the code would be minimal! The only thing is that `vtkStructuredDataPlaneCutter` has `GeneratePolygonsOff` instead of `GenerateTrianglesOff` and `SetPlane` instead of `SetCutFunction`.

### Links to VTK Documentation, Examples, or Class Definitions.

https://vtk.org/doc/nightly/html/classvtkStructuredDataPlaneCutter.html#details

**Comment by MatthewFlamm** [comment]
From the docs, it looks like this only works with 

> fast plane cutting of vtkImageData/vtkRectilinearGrid/vtkStructuredGrid

But maybe we should override this method for those DataSets in pyvista.

**Comment by gbene** [comment]
Yes exactly what I was thinking of. Since these kind of datasets can be pretty dense and computationally intensive for the classic vtkCutter maybe having an internal switch could be useful!
